httpie/input.py

```
@@ -0,0 +1,693 @@
+        if item.sep == SEP_HEADERS:
+            target = headers
+        elif item.sep == SEP_QUERY:
+            target = params
+        elif item.sep == SEP_FILES:
+            try:
+                with open(os.path.expanduser(value), 'rb') as f:
+                    value = (os.path.basename(value),
+                             BytesIO(f.read()),
+                             get_content_type(value))
+            except IOError as e:
+                raise ParseError('"%s": %s' % (item.orig, e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)